query-engine/connectors/sql-query-connector/src/filter_conversion.rs

```diff
@@ -435,6 +557,15 @@ fn default_scalar_filter(
             }
             _ => comparable.not_in_selection(convert_values(fields, values)),
         },
+
+        ScalarCondition::JsonCompare(json_compare) => {
+            return convert_json_filter(
```

Comment:
Early return here is not great, I much rather prefer to separate the code paths. You can choose to match on json early and call into a different function for the other filters, for example.